sentry-android-navigation/src/main/java/io/sentry/android/navigation/SentryNavigationListener.kt

```diff
@@ -0,0 +1,64 @@
+package io.sentry.android.navigation
+
+import android.os.Bundle
+import androidx.navigation.NavController
+import androidx.navigation.NavDestination
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.SentryLevel.INFO
+import io.sentry.TypeCheckHint
+import java.lang.ref.WeakReference
+
+class SentryNavigationListener @JvmOverloads constructor(
+    private val hub: IHub = HubAdapter.getInstance()
+) : NavController.OnDestinationChangedListener {
+
+    private var previousDestinationRef: WeakReference<NavDestination>? = null
+    private var previousArgs: Bundle? = null
```

Comment:
Should the `Bundle` also be wrapped with `WeakReference`?

Reply:
nope, because it doesn't hold any reference to activity/context and it's just an android-specific version of `Map`, so we are good here